pages/tests/message-passing.js

```diff
@@ -1,197 +1,164 @@
-  
-  // Pipeline setup
-
-  const computePipeline = device.createComputePipeline({
-    layout: device.createPipelineLayout({
-      bindGroupLayouts: [bindGroupLayout]
-    }),
-    compute: {
-      module: messagePassingModule,
-      entryPoint: "main"
-    }
-  });
-
-  // Commands submission
-
-  const commandEncoder = device.createCommandEncoder();
-
-  const passEncoder = commandEncoder.beginComputePass();
-  passEncoder.setPipeline(computePipeline);
-  passEncoder.setBindGroup(0, bindGroup);
-  passEncoder.dispatch(2);
-  passEncoder.endPass();
-
-  const resultsReadBuffer = device.createBuffer({
-    size: 8,
-    usage: GPUBufferUsage.COPY_DST | GPUBufferUsage.MAP_READ
-  });
-
-  commandEncoder.copyBufferToBuffer(
-    resultsBuffer,
-    0,
-    resultsReadBuffer,
-    0,
-    8
-  );
-
-  // Submit GPU commands.
-  const gpuCommands = commandEncoder.finish();
-  device.queue.submit([gpuCommands]);
-
-  // Read buffer.
-  await resultsReadBuffer.mapAsync(GPUMapMode.READ);
-  const arrayBuffer = resultsReadBuffer.getMappedRange();
-  console.log(new Uint32Array(arrayBuffer));
-  return new Uint32Array(arrayBuffer);
-}
-
+import React,{ useState } from 'react';
+//import Script from 'next/script';
+import Layout from '../../components/layout';
+import * as ReactBootStrap from 'react-bootstrap';
+const results = [];
 export default function StoreBuffer() {
-  try {
-    const p = doMessagePassing();
-    if (p instanceof Promise) {
-      p.catch((err) => {
-        console.error(err);
-        setError(err);
-      });
-    }
-  } catch (err) {
-    console.error(err);
-    setError(err);
+  const [iteration, setiteration] = useState(null);
+  const [loading, setLoading] = useState(false);
+  const [result, setResult] = useState(null);
+  const [pesudoSwitch, setPesudo]=useState("");
+  const [sourceSwitch, setSource]=useState("is-hidden");
+  const [pesudoActive, setPesudoActive]=useState("is-active");
+  const [sourceActive, setSourceActive]=useState("");
+  
+  async function doMessagePassing(num) {
+    setLoading(true);
+    let sum = 1;
+   for(let i = 0; i < num; i ++){
+     sum +=sum * i;
+   }
+   console.log("this is result "+sum);
+   setTimeout(() => {
+    setLoading(false);
+   }, 3000);
+   setResult(sum);
+   return sum;
   }
+  // try {
+  //   const p = doMessagePassing(iteration);
+  //   if (p instanceof Promise) {
+  //     p.catch((err) => {
+  //       console.error(err);
+  //       setError(err);
+  //     });
+  //   }
+  // } catch (err) {
+  //   console.error(err);
+  //   setError(err);
+  // }
   return (
-      <>
-        <h1>Message Passing</h1>
-        <h2>
-          <Link href="/">
-            <a>Back to home</a>
-          </Link>
-        </h2>
-      </>
-    )
+  <>
+    <div class="columns">
+      <div class="column">
+        <h1 class="testName">Message Passing</h1>
+          <h2 class="testDescription">Description goes here</h2>
+      </div>
+    </div>
+    <div class=" columns">
```

Comment:
I'm getting an error here:
`Warning: Invalid DOM property `class`. Did you mean `className`?`

Should this be `className`?